File: app/src/main/java/home/oleg/placesnearme/common/provider/ResourceProvider.java
package home.oleg.placesnearme.common.provider;

import android.graphics.drawable.Drawable;
import android.support.annotation.ColorRes;
import android.support.annotation.DrawableRes;
import android.support.annotation.Nullable;
import android.support.annotation.StringRes;

/**
 * Created by Oleg Sheliakin on 28.08.2018.
 * Contact me by email - olegsheliakin@gmail.com
 */
public interface ResourceProvider {

    @Nullable
    Drawable getDrawable(@DrawableRes int resId);

    @Nullable
    String getString(@StringRes int resId);

    int getColor(@ColorRes int resId);

}

File: core-presentation/src/main/java/home/oleg/placesnearme/core_presentation/provider/ResourceProvider.kt
package home.oleg.placesnearme.core_presentation.provider

import android.graphics.drawable.Drawable

import androidx.annotation.ColorRes
import androidx.annotation.DrawableRes
import androidx.annotation.Nullable
import androidx.annotation.StringRes

/**
 * Created by Oleg Sheliakin on 28.08.2018.
 * Contact me by email - olegsheliakin@gmail.com
 */
interface ResourceProvider {

    fun getDrawable(@DrawableRes resId: Int): Drawable?

    fun getString(@StringRes resId: Int): String

    fun getColor(@ColorRes resId: Int): Int

}
